File: snake.py
from position import Position

class Snake:
    DIRECTIONS = {"w": Position(0, -1), "a": Position(-1, 0),
                  "s": Position(0, 1), "d": Position(1, 0)}

    def __init__(self):
        self.head = Position(0, 0)
        self.body = [self.head]
        self.direction = self.DIRECTIONS["d"]

    def set_direction(self, direction):
        if direction in self.DIRECTIONS:
            self.direction = self.DIRECTIONS[direction]

    def move(self):
        self.head = self.head + self.direction
        self.body.insert(0, self.head)
        return self.body.pop()

    def grow(self):
        self.body.append(self.body[-1])

File: snake.go
package main

type Snake struct {
	Body []Point
	Dir  Direction
}

func NewSnake() *Snake {
	return &Snake{
		Body: []Point{{5, 5}},
		Dir:  Right,
	}
}

func (s *Snake) Move() {
	// Move the snake in the current direction
}

func (s *Snake) Grow() {
	// Add a new segment to the snake's body
}

func (s *Snake) CollidesWith(p Point) bool {
	// Check if the snake collides with the given point
}

func (s *Snake) CollidesWithSelf() bool {
	// Check if the snake collides with itself
}
